corehq/apps/domain/decorators.py

```diff
@@ -100,21 +101,28 @@ def call_view(): return view_func(req, domain_name, *args, **kwargs)
                 return TemplateResponse(request=req, template='two_factor/core/otp_required.html', status=403)
             elif not _can_access_project_page(req):
                 return _redirect_to_project_access_upgrade(req)
-            else:
-                return call_view()
-        elif user.is_superuser:
+            return call_view()
+
+        if user.is_superuser:
             if domain_obj.restrict_superusers and not _page_is_whitelisted(req.path, domain_obj.name):
                 from corehq.apps.hqwebapp.views import no_permissions
                 msg = "This project space restricts superuser access.  You must request an invite to access it."
                 return no_permissions(req, message=msg)
             if not _can_access_project_page(req):
                 return _redirect_to_project_access_upgrade(req)
             return call_view()
-        elif couch_user.is_web_user() and domain_obj.allow_domain_requests:
+
+        from corehq.apps.linked_domain.dbaccessors import get_domain_master_link
+        master_link = get_domain_master_link(domain_obj.name)
+        if master_link and not master_link.is_remote:
+            if ENTERPRISE_LINKED_DOMAINS.enabled(master_link.master_domain):
+                return call_view()
```

Comment:
I believe this is missing a check for membership/permission

Reply:
updated in cdd8e0fc93c